src/meta-srv/src/table_meta_alloc.rs

```diff
@@ -66,7 +71,23 @@ impl TableMetadataAllocator for MetaSrvTableMetadataAllocator {
         )
         .await
         .map_err(BoxedError::new)
-        .context(meta_error::ExternalSnafu)
+        .context(meta_error::ExternalSnafu)?;
+
+        let region_topics = match self.wal_meta_allocator.wal_provider() {
+            WalProvider::RaftEngine => vec![],
+            WalProvider::Kafka => {
+                // Each region gets assigned one topic.
+                self.wal_meta_allocator
+                    .try_alloc_topics(region_routes.len())
+                    .await?
+            }
+        };
```

Comment:
There are too many matches. Why not `let region_config = self.wal_meta_allocator.allocate()`

Reply:
Those codes are removed now.